mac

```diff
@@ -181,16 +181,36 @@ install_asdf_language() {
   fi
 }
 
+install_asdf_language_version() {
+  local language="$1"
+  local version="$2"
+
+  if ! asdf list "$language" | grep -Fq "$version"; then
+    asdf install "$language" "$version"
+    asdf global "$language" "$version"
+  fi
+}
+
+start_postgres_server() {
+  if ! pg_ctl status | grep -Fq "server is running"; then
+    pg_ctl start
+  fi
+}
+
 fancy_echo "Installing latest Ruby..."
-install_asdf_language "ruby"
+install_asdf_language_latest "ruby"
 gem update --system
 gem_install_or_update "bundler"
 number_of_cores=$(sysctl -n hw.ncpu)
 bundle config --global jobs $((number_of_cores - 1))
 
 fancy_echo "Installing latest Node..."
 bash "$HOME/.asdf/plugins/nodejs/bin/import-release-team-keyring"
-install_asdf_language "nodejs"
+install_asdf_language_latest "nodejs"
+
+fancy_echo "Installing Postgres 9.6.9..."
+install_asdf_language_version "postgres" "9.6.9"
+start_postgres_server
```

Comment:
If on different versions of Postgres, will we need to stop the service?

Reply:
Does `asdf` have postgres work with with `launchctl` or whatever on the Mac makes things autostart for you?

Reply:
> If on different versions of Postgres, will we need to stop the service?

From my limited experience with postgres, it looks like you do have to stop the service before switching versions. However, I don't think this is an `asdf` problem, I think it's a problem with any installation of multiple versions of postgres.

Reply:
> Does asdf have postgres work with launchctl or whatever on the Mac makes things autostart for you?

I have absolutely no idea, I would need to look into that.

Reply:
I know that by installing postgres through `asdf` the way this script would, the server would not start. I'm not sure if there's some sort of configuration you could do to make this happen; that's what I would need to research.